parser/charset/encoding_gbk.go

```diff
@@ -42,6 +44,35 @@ func (e *encodingGBK) Tp() EncodingTp {
 	return EncodingTpGBK
 }
 
+// Foreach implements Encoding interface.
+func (e *encodingGBK) Foreach(src []byte, op Op, fn func(from, to []byte, ok bool) bool) {
```

Comment:
I think it is a little bit elegant to hijack the `simplifiedchinese.GBK`:

https://gist.github.com/tangenta/89f36e22c1a7cfffbf02f75c6d9c1f20